src/main/java/core/basesyntax/operations/BalanceOperation.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void apply(Storage storage, FruitTransaction transaction) {
+        if (storage == null) {
+            throw new IllegalArgumentException("Storage cannot be null.");
+        }
+        if (transaction == null) {
+            throw new IllegalArgumentException("Transaction cannot be null.");
+        }
+        if (transaction.getQuantity() < 0) {
+            throw new IllegalArgumentException("Quantity must be non-negative.");
+        }
+
+        storage.updateFruit(transaction.getFruit(), transaction.getQuantity());

```

Comment:
The `apply` method should not directly interact with the `Storage` class. Instead, consider following the SOLID principles and make your services independent. You could, for example, return the result of the operation and then have that result applied to the storage in another part of your application, such as the main method or a dedicated service.